AdyenActions/AdyenActionComponent.swift

```diff
@@ -130,42 +118,82 @@ public final class AdyenActionComponent: ActionComponent, Localizable {
             delegate?.didFail(with: ComponentError.paymentMethodNotSupported, from: self)
             return
         }
-        weChatPaySDKActionComponent = classObject.init(apiContext: apiContext)
-        weChatPaySDKActionComponent?._isDropIn = _isDropIn
-        weChatPaySDKActionComponent?.delegate = delegate
-        weChatPaySDKActionComponent?.handle(action)
+        
+        let weChatPaySDKActionComponent = classObject.init(apiContext: apiContext)
+        weChatPaySDKActionComponent._isDropIn = _isDropIn
+        weChatPaySDKActionComponent.delegate = delegate
+        weChatPaySDKActionComponent.handle(action)
+        
+        currentActionComponent = weChatPaySDKActionComponent
     }
     
     private func handle(_ action: AwaitAction) {
-        let component = AwaitComponent(apiContext: apiContext, style: awaitComponentStyle)
+        let component = AwaitComponent(apiContext: apiContext, style: style.awaitComponentStyle)
         component._isDropIn = _isDropIn
         component.delegate = delegate
         component.presentationDelegate = presentationDelegate
         component.localizationParameters = localizationParameters
         
         component.handle(action)
-        awaitComponent = component
+        currentActionComponent = component
     }
     
     private func handle(_ action: VoucherAction) {
-        let component = VoucherComponent(apiContext: apiContext, style: voucherComponentStyle)
+        let component = VoucherComponent(apiContext: apiContext, style: style.voucherComponentStyle)
         component._isDropIn = _isDropIn
         component.delegate = delegate
         component.presentationDelegate = presentationDelegate
         component.localizationParameters = localizationParameters
 
         component.handle(action)
-        voucherComponent = component
+        currentActionComponent = component
     }
     
     private func handle(_ action: QRCodeAction) {
-        let component = QRCodeComponent(apiContext: apiContext, style: QRCodeComponentStyle())
+        let component = QRCodeComponent(apiContext: apiContext, style: style.qrCodeComponentStyle)
         component._isDropIn = _isDropIn
         component.delegate = delegate
         component.presentationDelegate = presentationDelegate
         component.localizationParameters = localizationParameters
         
         component.handle(action)
-        qrCodeComponent = component
+        currentActionComponent = component
     }
 }
+
+public extension AdyenActionComponent {
```

Comment:
This extension could be in it's own file

Reply:
yeah i was just in a hurry :)